controllers/argocd/applicationset.go

```diff
@@ -328,6 +356,12 @@ func (r *ReconcileArgoCD) reconcileApplicationSetRole(cr *argoprojv1a1.ArgoCD) (
 		if !errors.IsNotFound(err) {
 			return nil, fmt.Errorf("failed to reconcile the role for the service account associated with %s : %s", role.Name, err)
 		}
+		if applicationSetDisabled {
+			if err := r.Client.Delete(context.TODO(), role); err != nil {
```

Comment:
can we have a `log.Info` here that mentions that the applicationSet role is being deleted because applicationSet was disabled?